reana_workflow_controller/rest/workflows.py

```diff
@@ -254,7 +255,9 @@ def get_workflows(paginate=None):  # noqa
             workflow_status = [RunStatus[status] for status in status_list.split(",")]
             query = query.filter(Workflow.status.in_(workflow_status))
         if workflow:
-            query = query.filter(Workflow.name == workflow)
+            query = query.filter(
+                or_(Workflow.name == workflow, Workflow.id_ == workflow)
```

Comment:
I think this breaks the filter by workflow name. For instance, this call (`https://localhost:30443/api/workflows?workflow_id_or_name=roofit`) is giving an error to me, can you reproduce the same? 

It looks like `query.filter(Workflow.id_ == workflow)` doesn't like comparisons with regular strings.

```python
>>> user.workflows.filter(Workflow.id_ == 'foo').count()
[StatementError: StatementError('(builtins.ValueError) bytes is not a 16-char string')](http://localhost:31984/debug/session/140507376465536)
```

Reply:
Amazing find, thanks! Unexpected behavior while filtering uuids.. I've added a quick check to distinguish queries for uuids and workflow names

Reply:
Works well now! 👍 Strange behavior indeed.